core/server/common/src/main/java/alluxio/master/PrimarySelectorClient.java

```diff
@@ -158,12 +212,14 @@ public void takeLeadership(CuratorFramework client) throws Exception {
         .forPath(mLeaderFolder + mName);
     LOG.info("{} is now the leader.", mName);
     try {
+      mLeaderZkSessionId = client.getZookeeperClient().getZooKeeper().getSessionId();
       waitForState(State.SECONDARY);
     } finally {
       LOG.warn("{} relinquishing leadership.", mName);
       LOG.info("The current leader is {}", mLeaderSelector.getLeader().getId());
       LOG.info("All participants: {}", mLeaderSelector.getParticipants());
```

Comment:
Can we log what we think the zk session id was? Would that be useful info?

Reply:
What it was will be logged as I added a trace for that already.